modules/widgets/simple-payments.php

```diff
@@ -463,24 +464,24 @@ function update( $new_instance, $old_instance ) {
 		function form( $instance ) {
 			$instance = wp_parse_args( $instance, $this->defaults() );
 
-			$product_posts = get_posts( array(
-				'numberposts' => 100,
-				'orderby' => 'date',
-				'post_type' => Jetpack_Simple_Payments::$post_type_product,
-				'post_status' => 'publish',
-			 ) );
+			$jetpack_simple_payments = Jetpack_Simple_Payments::getInstance();
+			if ( $jetpack_simple_payments->is_enabled_jetpack_simple_payments() ) {
+				$product_posts = get_posts( array(
+					'numberposts' => 100,
+					'orderby' => 'date',
+					'post_type' => Jetpack_Simple_Payments::$post_type_product,
+					'post_status' => 'publish',
+				) );
 
-			require( dirname( __FILE__ ) . '/simple-payments/form.php' );
+				require( dirname( __FILE__ ) . '/simple-payments/form.php' );
+			} else {
+				require( dirname( __FILE__ ) . '/simple-payments/admin-warning.php' );
+			}
```

Comment:
maybe an early-abort here would be nice?

```php
if ( ! $jetpack_simple_payments->is_enabled_jetpack_simple_payments() ) {
	require( dirname( __FILE__ ) . '/simple-payments/admin-warning.php' );
	return;
}

$product_posts = …
```

would at least make the differ simpler I think.